spec/intelligent_foods/resources/order_spec.rb

```diff
@@ -70,6 +70,25 @@
       expect(result[:items]).to eq(expected_output)
     end
 
+    it "assigns the reference id as a string in the request body" do
+      recipient = build(:recipient)
+      menu = build(:menu, id: "2023-01-01")
+      order_item = build(:order_item, quantity: 2)
+      external_id = 1234
+      order = IntelligentFoods::Order.new(menu: menu,
+                                          recipient: recipient,
+                                          items: [order_item],
+                                          external_id: 1234)
+      expected_output = external_id.to_s
+      body = build_order_response
+      response = build_response(body: body)
+      stub_api_response response: response
+
+      result = order.request_body
+
+      expect(result[:reference_id]).to eq(expected_output)
```

Comment:
i'm not sure if `eq` is the right one here. because it does auto casting. i know i say to never test types, but this might be a reasonable use case for this test

Reply:
or if there is a better way to ensure that it is in fact a string? or are we happy it just quacks like a string?